react/features/toolbar/components/ToolbarButton.web.js

```diff
@@ -0,0 +1,206 @@
+/* @flow */
+
+import React from 'react';
+
+import { translate } from '../../base/i18n';
+
+import UIUtil from '../../../../modules/UI/util/UIUtil';
+
+import AbstractToolbarButton from './AbstractToolbarButton';
+import { getButtonAttributesByProps } from '../functions';
+
+declare var APP: Object;
+declare var interfaceConfig: Object;
+
+/**
+ * Represents a button in Toolbar on React.
+ *
+ * @class ToolbarButton
+ * @extends AbstractToolbarButton
+ */
+class ToolbarButton extends AbstractToolbarButton {
+    _createRefToButton: Function;
+
+    _onClick: Function;
+
+    /**
+     * Toolbar button component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        ...AbstractToolbarButton.propTypes,
+        button: React.PropTypes.object.isRequired,
+        t: React.PropTypes.func
+    };
+
+    /**
+     * Initializes new ToolbarButton instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props: Object) {
+        super(props);
+
+        // Bind methods to save the context
+        this._createRefToButton = this._createRefToButton.bind(this);
+        this._onClick = this._onClick.bind(this);
+    }
+
+    /**
+     * Sets shortcut/tooltip
+     * after mounting of the component.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentDidMount() {
+        this._setShortcutAndTooltip();
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const {
+            button,
+            key,
+            t
+        } = this.props;
+        const attributes = getButtonAttributesByProps(button);
+        const popups = button.popups || [];
+
+        return (
+            <a
+                { ...attributes }
+                onClick = { this._onClick }
+                ref = { this._createRefToButton }>
+                { this._renderInnerElementsIfRequired() }
+                { this._renderPopups(popups) }
+                { t(key) }
+            </a>
+        );
+    }
+
+    /**
+     * Creates reference to current toolbar button.
+     *
+     * @param {HTMLElement} element - HTMLElement representing the toolbar
+     * button.
+     * @returns {void}
+     * @private
+     */
+    _createRefToButton(element): void {
+        this.button = element;
+    }
+
+    /**
+     * Wrapper on on click handler props for current button.
+     *
+     * @param {Event} event - Click event object.
+     * @returns {void}
+     * @private
+     */
+    _onClick(event) {
```

Comment:
the previous function has a flow annotation, can you please be more consistent adding them?